wcfsetup/install/files/lib/system/session/SessionHandler.class.php

```diff
@@ -147,18 +125,40 @@ class SessionHandler extends SingletonFactory {
 	 */
 	protected $usersOnlyPermissions = [];
 	
+	private const ACP_SESSION_LIFETIME = 7200;
+	private const GUEST_SESSION_LIFETIME = 7200;
+	private const USER_SESSION_LIFETIME = 86400 * 14;
+	
 	/**
 	 * Provides access to session data.
 	 * 
 	 * @param	string		$key
 	 * @return	mixed
 	 */
 	public function __get($key) {
-		if (isset($this->environment[$key])) {
-			return $this->environment[$key];
+		if ($key === 'sessionID') {
+			return $this->sessionID;
 		}
+		else if ($key === 'userID') {
+			return $this->user->userID;
+		}
+		// TODO: pageID, pageObjectID, parentPageID, parentPageObjectID
 		
-		return $this->session->{$key};
+		/** @deprecated	5.4 - These values can be retrieved more efficiently by directly using the methods in e.g. UserUtil. */
```

Comment:
Would it be helpful for IDE use if I define these as public properties and immediately mark them as deprecated?

Reply:
You could have just kept the `protected $enviroment` variable for this purpose, I don't see a benefit of moving it here?

Reply:
I moved the stuff to slim the whole property list down. I don't believe marking the `$environment` property as deprecated would have helped with the following:

```
WCF::getSession()->userAgent // accesses the magic getter
```

The deprecation of `->environment` cannot magically extend to `->userAgent`. However I realize that adding public properties prevents the getter from being called :thinking:

Reply:
Hacking around with a trait works according to: https://stackoverflow.com/a/51890536/782822